pylint/checkers/base/basic_checker.py

```diff
@@ -453,7 +463,20 @@ def visit_expr(self, node: nodes.Expr) -> None:
 
         # Warn W0133 for exceptions that are used as statements
         if isinstance(expr, nodes.Call):
-            inferred = utils.safe_infer(expr)
+            inferred = None
+            if self._heuristic_exception_detection is True:
+                name, _, _ = expr.as_string().partition("(")
+                exc_suffixes = ("Error", "Exc", "Exception", "Warning")
+                if any(name.endswith(suffix) for suffix in exc_suffixes):
+                    inferred = utils.safe_infer(expr)
```

Comment:
Could you try the regex filter for PascalCase here?
```
[^\W\da-z][^\W_]+$
```

Reply:
Possibly needs a leading caret (`^`) character?

(otherwise matches `laFoobar`, for example)

Reply:
Attribute-accesses would also need handling, I think (`Foo.Bar`).

Reply:
> Possibly needs a leading caret (`^`) character?
> 
> (otherwise matches `laFoobar`, for example)

You can use `re.Match`. That always starts at the beginning of the string.
https://docs.python.org/3/library/re.html#re.match

Reply:
> Attribute-accesses would also need handling, I think (`Foo.Bar`).

This should work
```py
CLASS_NAME_RGX.match(
    expr.func.as_string().rsplit(".", maxsplit=1)[-1]
)
```

Reply:
That's helpful, thank you - I'm reading up on how `CLASS_NAME_RGX` (and `PascalCaseStyle`) are used throughout the `pylint` codebase.

Reply:
To be honest I feel like 'call name starts with an uppercase character' would achieve good-enough heuristic filtering to allow inference to proceed - but until I'm able to replicate the performance degradation, I don't have much confidence that I'll be able to confirm that the issue is resolved.